### ISSUE TITLE
Slowdown over time to improve user experience

### ISSUE THREAD

**Issue by styggeulf** [EnterpriseQualityCoding/FizzBuzzEnterpriseEdition#334]
To ensure the best possible user experience, we need to hard-code a slowdown if FizzBuzz detects any hardware or software older than 18 months on the client side. We cannot allow users on legacy hardware to experience any adverse effect on their laptop batteries due to excessive load from our application.

The business case practically writes itself: A mere 10% reduction in support cases will allow us to cut at least 40% of our support staff and result in more shareholder value. We have run the numbers and due dilligence forces our hand - we could face lawsuits if we do not implement this code change ASAP.

I don't know any fancy code, but it could look something like this in BASIC:

```
10 [whatever the command for detecting hardware age for all components client side is]
15 [whatever the command for detecting software age for all related programmes cleint side is]
20 [generate some sort of aggregate score based on 10 and 15, let's call it [score]]
25 IF [score] > 18 months GOTO 40 ELSE GOTO 30
30 [don't slow down]
40 [slow down by 2% per day over 18 months of age]
50 END
```

So basically no more than 7 lines of code. Should be no more than half an hour's work for one non-gender-or-kin-specific coder monkey, the way I see it.

**Comment by styggeulf** [comment]
I am glad that you are on board with the idea, and I am happy to report that I can address your concerns.

Re. the code: Well, you can just copy-paste the BASIC code from my original post into Google Translate and select Java or COBOL, so I don't see the issue here (with all due respect).

Re. the slowdown: I have most certainly taken into account what happens when it goes above 100% - or rather, a certain Mr. Isaac Newton has. The third law of thermodynamics dictates that for every percentage point of slowdown above 100 on the client's machine, FizzBuzz Inc. gains a corresponding point in speed.

**Comment by StevenPolley** [comment]
Looks like this has been patched upstream by Intel, we can probably mark this as closed.
